bundles/org.openhab.binding.resol/src/main/java/org/openhab/binding/resol/handler/ResolThingHandler.java

```diff
@@ -0,0 +1,292 @@
+            } else if (pfv.getPacketFieldSpec().getType() == SpecificationFile.Type.DateTime) {
+                channelTypeUID = new ChannelTypeUID(ResolBindingConstants.BINDING_ID, "DateTime");
+            } else {
+                /* used for enums and the numeric types without unit */
+                channelTypeUID = new ChannelTypeUID(ResolBindingConstants.BINDING_ID, "None");
+            }
+
+            String acceptedItemType;
+
+            Thing thing = getThing();
+            switch (pfv.getPacketFieldSpec().getType()) {
+                case DateTime:
+                    acceptedItemType = "DateTime";
+                    break;
+                case WeekTime:
+                case Number:
+                    acceptedItemType = ResolChannelTypeProvider.itemTypeForUnit(pfv.getPacketFieldSpec().getUnit());
+                    break;
+                case Time:
+                default:
+                    acceptedItemType = "String";
+                    break;
+            }
+            Channel a = thing.getChannel(channelId);
+
+            if (a == null) {
+                /* channel doesn't exit, let's create it */
+                ThingBuilder thingBuilder = editThing();
+                ChannelUID channelUID = new ChannelUID(thing.getUID(), channelId);
+
+                if (pfv.getEnumVariant() != null) {
+                    /* create a state option channel */
+                    List<StateOption> options = new ArrayList<>();
+                    PacketFieldSpec ff = pfv.getPacketFieldSpec();
+                    Enum e = ff.getEnum();
+                    for (long l : e.getValues()) {
+                        EnumVariant v = e.getEnumVariantForValue(l);
+                        options.add(new StateOption(Long.toString(l), v.getText(lang)));
+                    }
+
+                    stateDescriptionProvider.setStateOptions(channelUID, options);
+
+                    Channel channel = ChannelBuilder.create(channelUID, "Number").withType(channelTypeUID)
+                            .withLabel(pfv.getName(lang)).build();
+
+                    thingBuilder.withChannel(channel).withLabel(thing.getLabel());
+                    updateThing(thingBuilder.build());
+                } else if (pfv.getRawValueDouble() != null) {
+                    /* a number channel */
+                    Channel channel = ChannelBuilder.create(channelUID, acceptedItemType).withType(channelTypeUID)
+                            .withLabel(pfv.getName(lang)).build();
+
+                    thingBuilder.withChannel(channel).withLabel(thing.getLabel());
+                    updateThing(thingBuilder.build());
+                }
+                logger.debug("Creating channel: {}", channelUID);
+            }
+
+            if (pfv.getEnumVariant() != null) {
+                /* update the enum / State channel */
+                setChannelValue(channelId, pfv.getRawValueLong()); // pfv.getEnumVariant().getText(lang));
```

Comment:
Can the comment be removed?

Reply:
it can not only, it shall be removed :grinning: